corehq/apps/reports/commtrack/data_sources.py

```diff
@@ -418,26 +432,30 @@ def get_data(self):
             for spoint_id, loc_id in spoint_loc_map.items():
                 loc = locations[loc_id]
 
-                form_ids = StockReport.objects.filter(
+                results = StockReport.objects.filter(
                     stocktransaction__case_id=spoint_id
-                ).exclude(
-                    date__lte=self.start_date
-                ).exclude(
-                    date__gte=self.end_date
```

Comment:
gotcha. second sql query will probably be faster if you can push the aggregation to the database.